File: app/src/main/java/io/github/alexeychurchill/stickynotes/activity/login/LoginFragment.java
package io.github.alexeychurchill.stickynotes.activity.login;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.ProgressBar;

import io.github.alexeychurchill.stickynotes.R;

/**
 * Login fragment
 */

public class LoginFragment extends Fragment implements View.OnClickListener {
    private EditText mETLogin;
    private EditText mETPassword;
    private ProgressBar mPBWait;
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_login, container, false);
        mETLogin = ((EditText) view.findViewById(R.id.etLogin));
        mETPassword = ((EditText) view.findViewById(R.id.etPassword));
        mPBWait = ((ProgressBar) view.findViewById(R.id.pbWait));
        Button btnLogin = ((Button) view.findViewById(R.id.btnLogin));
        if (btnLogin != null) {
            btnLogin.setOnClickListener(this);
        }
        return view;
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btnLogin:
                callLogin();
                break;
        }
    }

    private void callLogin() {
        //...
    }
}

File: app/src/main/java/io/github/alexeychurchill/stickynotes/account/LoginFragment.kt
package io.github.alexeychurchill.stickynotes.account

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import android.widget.Toast.LENGTH_LONG
import androidx.compose.ui.platform.ComposeView
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.lifecycleScope
import dagger.hilt.android.AndroidEntryPoint
import io.github.alexeychurchill.stickynotes.account.ui.LoginScreen
import io.github.alexeychurchill.stickynotes.core.StickyNotesTheme

/**
 * Login fragment
 */
@AndroidEntryPoint
class LoginFragment : Fragment() {
    interface OnLoggedInListener {
        fun onLoggedIn()
    }

    private val viewModel: LoginViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        lifecycleScope.launchWhenCreated {
            viewModel.onLogin.collect {
                (requireActivity() as? OnLoggedInListener)?.onLoggedIn()
            }
        }

        lifecycleScope.launchWhenCreated {
            viewModel.error.collect {
                Toast.makeText(requireContext(), it, LENGTH_LONG).show()
            }
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return ComposeView(requireContext())
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val composeView = view as? ComposeView ?: return

        composeView.setContent {
            StickyNotesTheme {
                LoginScreen(viewModel)
            }
        }
    }
}
